astropy/table/column.py

```diff
@@ -44,77 +45,23 @@ def _auto_names(n_cols):
      np.isfinite, np.isinf, np.isnan, np.sign, np.signbit])
 
 
-COLUMN_ATTRS = set(['name', 'unit', 'dtype', 'format', 'description', 'meta', 'parent_table'])
-
-def col_setattr(col, attr, value):
-    """
-    Set one of the column attributes.
-
-    Warning: this function is subject to change or removal.
-    """
-    if attr not in COLUMN_ATTRS:
-        raise AttributeError("attribute must be one of {0}".format(COLUMN_ATTRS))
-
-    # The unit and dtype attributes are considered universal and do NOT get
-    # stored in _astropy_column_attrs.  For BaseColumn instances use the usual setattr.
-    if isinstance(col, BaseColumn):
-        setattr(col, attr, value)
-    else:
-        # If no _astropy_column_attrs or it is None then convert to dict
-        if getattr(col, '_astropy_column_attrs', None) is None:
-            col._astropy_column_attrs = {}
-        if attr == 'parent_table':
-            value = None if value is None else weakref.ref(value)
-        col._astropy_column_attrs[attr] = value
-
-def col_getattr(col, attr, default=None):
-    """
-    Get one of the column attributes
-
-    Warning: this function is subject to change or removal.
-    """
-    if attr not in COLUMN_ATTRS:
-        raise AttributeError("attribute must be one of {0}".format(COLUMN_ATTRS))
-
-    # The unit and dtype attributes are considered universal and do NOT get
-    # stored in _astropy_column_attrs.  For BaseColumn instances use the usual setattr.
-    if (isinstance(col, BaseColumn) or
-            (isinstance(col, Quantity) and attr in ('dtype', 'unit'))):
-        value = getattr(col, attr, default)
-    else:
-        # If col does not have _astropy_column_attrs or it is None (meaning
-        # nothing has been set yet) then return default, otherwise look for
-        # the attribute in the astropy_column_attrs dict.
-        if getattr(col, '_astropy_column_attrs', None) is None:
-            value = default
-        else:
-            value = col._astropy_column_attrs.get(attr, default)
-
-        # Weak ref for parent table
-        if attr == 'parent_table' and callable(value):
-            value = value()
-
-        # Mixins have a default dtype of Object if nothing else was set
-        if attr == 'dtype' and value is None:
-            value = np.dtype('O')
-
-    return value
-
 def _col_update_attrs_from(newcol, col, exclude_attrs=['name', 'parent_table']):
     """
-    Update _astropy_column_attrs from mixin `col` to `newcol`.  Does nothing
-    for BaseColumn cols
+    Copy info from mixin `col` to `newcol`.  Does nothing for BaseColumn cols.
 
     Warning: this function is subject to change or removal.
     """
     if isinstance(newcol, BaseColumn):
         return
 
-    attrs = COLUMN_ATTRS - set(exclude_attrs)
+    if not hasattr(newcol, 'info'):
+        newcol.__class__.info = DataInfo(BaseInfo)
```

Comment:
I meant that for instances, we don't really want to touch their class at all, so why not just do what `DataInfo` does (effectively), stick on an `info` attribute. Of course, this would not be a property and thus could be overwritten, but I think that disadvantage is less than sticking something on a parent class. (I just can't bear the thought of working on some new fancy class and suddenly a property appears on it that isn't even in the definition -- how does one debug that?).